### PR TITLE
Added binary compute-online-feats.

### PR DESCRIPTION
**Pull request by luitjens**
This binary computes features and ivectors using the online2 recipe.

### CODE REVIEW

**Comment by luitjens** [comment]
Dan any idea why this check isn't running? It looks like it is just stuck.

**Comment by danpovey** [comment]
Likely just temporary outage on the test platform.

File: src/online2bin/compute-online-feats.cc [hunk 1]
```diff
@@ -0,0 +1,119 @@
+// online2/compute-online-feats.cc
+//
```
Comment (danpovey) [inline src/online2bin/compute-online-feats.cc]: Is this binary substantially different from  online2-wav-dump-features?

**Comment by luitjens** [comment]
Looks like the main differences are

it outputs the ivector features and does not handle speaker2utt.   Mostly
this binary is used to validate the cuda implementation of the feature
pipeline.  It is nice to have a binary we can compare against.  If you
would like we could move it under cudafeatbin though that may not make
sense since it does not use CUDA.

**Comment by danpovey** [comment]
I think I would prefer if you just modify the existing program.  Will avoid
bulking things up, the compiled
set of programs is already too large.   How about modifying the existing
program so that if you give it a 4th argument it will have a usage like
yours?  (Would still take a spk2utt-rspecifier that
would, in your normal usage, be a one to one map).

Suggested modified usage message:
```
        "Usage (1):\n"
        " online2-wav-dump-features [options] <spk2utt-rspecifier>
<wav-rspecifier> <feature-wspecifier>\n"
        "   The spk2utt-rspecifier can just be <utterance-id>
<utterance-id> if\n"
        "    you want to generate features utterance by utterance.\n"
        "Usage (2):\n"
        " online2-wav-dump-features [options] <spk2utt-rspecifier>
<wav-rspecifier> <feature-wspecifier> <ivector-wspecifier>\n"
         "   In this usage pattern, the features and ivectors will be
dumped separately (the ivectors\n"
         "   will be dumped as vectors after being taken from the last
frame of the last utterance\n"
         "   of each speaker\n"
        "Usage (3):\n"
        "     online2-wav-dump-features [options]
--print-ivector-dim=true\n"
        "See steps/online/nnet2/{dump_nnet_activations,get_egs.sh} for
examples.\n";
```

If you have testing scripts that you use, it might be nice to have them
checked in somewhere.

**Comment by luitjens** [comment]
ok,  i think we can close this and i'll try to get back and add the features to the existing binary.  This is low priority since it doesn't impact functionality at all.

**Comment by danpovey** [comment]
ok.

**Pull request closed without merging**